### DISCUSSION TITLE
Why the hell PNPM is using this much if internet bandwidth when it's not even downloading anything new!!!

### DISCUSSION THREAD

**Discussion by babakfp** [pnpm/pnpm#7457] [Q&A] [answered]
![image](https://github.com/pnpm/pnpm/assets/44144724/d6bb8010-8795-436f-b1bb-fa4070b24f8f)

**Comment by zkochan** [comment]
Because the metadata of every package is downloaded. If you are OK that you sometimes won't fetch the latest versions of dependencies, you can set the `prefer-offline` setting to `true`.

**Reply by babakfp** [reply]
Thanks

Is there a way to set this globally? So, no matter what project I'm doing the installation... I don't want to use a config file or use it as a cli option. Can I set it for other package managers too?

Using as little bandwidth is important to me because the internet is expensive in my country...

---

My other question is, wouldn't it be better if we had static and strict versions, instead of `^` and `~`? So, we could not have the lock file, we could not worry about using more bandwidth because of metadata, we could not worry about things breaking in minor version changes, and...

**Reply by zkochan** [reply, marked as answer]
You can set the setting globally via

```
pnpm config set prefer-offline true
```

> My other question is, wouldn't it be better if we had static and strict versions, instead of ^ and ~? So, we could not have the lock file, we could not worry about using more bandwidth because of metadata, we could not worry about things breaking in minor version changes, and...

You can do that with your own packages but the node.js ecosystem usually uses version ranges.

**Reply by babakfp** [reply]
Thanks 🌸
